src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -170,7 +170,8 @@ public IJsonTypeInfoResolver TypeInfoResolver
             [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
             get
             {
-                return _typeInfoResolver ?? DefaultJsonTypeInfoResolver.RootDefaultInstance();
+                InitializeForReflectionSerializer();
```

Comment:
this is not required change but it makes more sense since we're already doing unsafe stuff we might as well initialize it

Reply:
Hmm, this is modifying the options state by virtue of us calling a property getter. It would opt in reflection-based fallback if I fetched the resolver, _even if the options instance is tied to JsonSerializerContext_.

We're trying to undo this behavior for the options-based serialization APIs, so this feels like a step back.

Reply:
Is this related to the other change in any way?